packages/experiment-tag/src/experiment.ts

```diff
@@ -22,13 +22,18 @@ import {
   urlWithoutParamsAndAnchor,
   UUID,
   concatenateQueryParamsOf,
+  matchesUrl,
 } from './util';
 
 const appliedInjections: Set<string> = new Set();
 const appliedMutations: MutationController[] = [];
 let previousUrl: string | undefined = undefined;
+// Cache to track exposure for the current URL, should be cleared on URL change
+let urlExposureCache: { [url: string]: { [key: string]: string | undefined } };
 
 export const initializeExperiment = (apiKey: string, initialFlags: string) => {
+  // Clear the cache on initialization
+  urlExposureCache = {};
```

Comment:
Why? If anything this should be below the check for globalScope.webExperiment

Reply:
This was to take care of some unexpected behavior in unit tests.